### PR TITLE
[Merged by Bors] - Activation lint

### PR DESCRIPTION
**Pull request by antonlerner**
## Motivation
Better linting of the Activation, Miner and State dirs

still needs more complex comments on private methods

I've also did not comment and lint :
state/processor.go
nipst related files
post related files

### CODE REVIEW

File: activation/activation.go [hunk 1]
```diff
@@ -12,106 +12,109 @@ import (
...
-	Epoch(l types.LayerID) types.EpochId
-}
-
-type Broadcaster interface {
+type broadcaster interface {
 	Broadcast(channel string, data []byte) error
 }
 
-type PoETNumberOfTickProvider struct {
+type poETNumberOfTickProvider struct {
```
Comment (noamnelke) [inline activation/activation.go]: this is very confusing... Let's consider poet a word and make the `ET` lowercase?

File: activation/activation.go [hunk 2]
```diff
@@ -12,106 +12,109 @@ import (
 	"sync/atomic"
 )
 
+// AtxProtocol is the protocol id for broadcasting atxs over gossip
 const AtxProtocol = "AtxGossip"
 
 var activesetCache = NewActivesetCache(1000)
 
-type MeshProvider interface {
+// meshProviderMock
```
Comment (noamnelke) [inline activation/activation.go]: remove, since you made this private

File: activation/activation.go [hunk 3]
```diff
@@ -12,106 +12,109 @@ import (
...
+	// InitIdle status means that an init file does not exist and is not prepared
 	InitIdle = 1 + iota
+	// InitInProgress status means that an init file preparation is now in progress
 	InitInProgress
+	// InitDone status indicates there is a prepared init file
 	InitDone
 )
 
+// Builder struct is the struct that orchestrates the creation of activation transactions
+// it is responsible for
```
Comment (noamnelke) [inline activation/activation.go]: incomplete comment

File: activation/activation.go [hunk 4]
```diff
@@ -143,10 +146,13 @@ func (b *Builder) Stop() {
 	close(b.stop)
 }
 
+// SignAtx signs the atx atx and assigns the signature into atx.Sig
```
Comment (noamnelke) [inline activation/activation.go]:
```suggestion
// SignAtx signs the atx and assigns the signature into atx.Sig
```

File: activation/activation.go [hunk 5]
```diff
@@ -217,6 +223,8 @@ func (b *Builder) buildNipstChallenge() error {
 	return nil
 }
 
+// StartPost initiates post commitment generation process. it returns an error if a process is already on progress or
```
Comment (noamnelke) [inline activation/activation.go]:
```suggestion
// StartPost initiates post commitment generation process. It returns an error if a process is already in progress or
```

File: activation/activation_test.go [hunk 6]
```diff
@@ -82,7 +82,7 @@ func (mpm *MeshProviderMock) LatestLayer() types.LayerID {
 
 type NetMock struct {
 	lastTransmission []byte
-	atxDb            ATXDBProvider
+	atxDb            atxDBProvider
```
Comment (noamnelke) [inline activation/activation_test.go:85]: Should be `atxDB` (capital b), I guess?

File: activation/atxdb.go [hunk 7]
```diff
@@ -40,15 +40,17 @@ type atxChan struct {
 	listeners int
 }
 
+// ActivationDb hold the atxs received from all nodes and their validity status
+// it also stores identifications for all nodes e.g the coupling between ed id and bls id
 type ActivationDb struct {
```
Comment (noamnelke) [inline activation/atxdb.go:45]: `ActivationDB`?

File: activation/atxdb.go [hunk 8]
```diff
@@ -57,9 +59,11 @@ type ActivationDb struct {
 	atxChannels       map[types.AtxId]*atxChan
 }
 
-func NewActivationDb(dbstore database.Database, idstore IdStore, meshDb *mesh.MeshDB, layersPerEpoch uint16, nipstValidator NipstValidator, log log.Log) *ActivationDb {
+// NewActivationDb creates a new struct of type ActivationDb, this struct will hold the atxs received from all nodes and
+// their validity
+func NewActivationDb(dbstore database.Database, idstore idStore, meshDb *mesh.MeshDB, layersPerEpoch uint16, nipstValidator nipstValidator, log log.Log) *ActivationDb {
```
Comment (noamnelke) [inline activation/atxdb.go:64]: `NewActivationDB`?

File: activation/cache.go [hunk 9]
```diff
@@ -6,10 +6,12 @@ import (
 	"github.com/spacemeshos/go-spacemesh/common/types"
 )
 
+// ActivesetCache holds and lru cache of the active set size for a view hash.
```
Comment (noamnelke) [inline activation/cache.go]:
```suggestion
// ActivesetCache holds an lru cache of the active set size for a view hash.
```

File: activation/cache.go [hunk 10]
```diff
@@ -31,10 +35,14 @@ func (bc ActivesetCache) Get(view types.Hash12) (uint32, bool) {
 	return blk, true
 }
 
+// AtxCache  holds an lru cache of ActivationTxHeader structs of recent atx used to calculate active set size
+// ideally this cache will hold the atxs created in latest epoch, on which mose of active set size calculation will be
+// performed
```
Comment (noamnelke) [inline activation/cache.go]:
```suggestion
// AtxCache holds an lru cache of ActivationTxHeader structs of recent atx used to calculate active set size
// ideally this cache will hold the atxs created in latest epoch, on which most of active set size calculation will be
// performed
```

File: activation/cache.go [hunk 11]
```diff
@@ -43,10 +51,13 @@ func NewAtxCache(size int) AtxCache {
 	return AtxCache{Cache: cache}
 }
 
+// Add adds an activationTxHeader to cache
 func (bc *AtxCache) Add(id types.AtxId, atxHeader *types.ActivationTxHeader) {
 	bc.Cache.Add(id, atxHeader)
 }
 
+// Get gets the corresponding Atx header to the given id id, it also returns a boolean to indicate whether the item
```
Comment (noamnelke) [inline activation/cache.go]:
```suggestion
// Get gets the corresponding Atx header to the given id, it also returns a boolean to indicate whether the item
```

File: miner/block_builder.go [hunk 12]
```diff
@@ -20,81 +20,87 @@ import (
...
 
-const DefaultGasLimit = 10
-const DefaultFee = 1
+const defaultGasLimit = 10
+const defaultFee = 1
 
+// IncomingTxProtocol is the protocol identifier for tx received by gossip that is used by the p2p
 const IncomingTxProtocol = "TxGossip"
 
+// AtxsPerBlockLimit indicates the maximum number of atxs a block can refrence
```
Comment (noamnelke) [inline miner/block_builder.go]:
```suggestion
// AtxsPerBlockLimit indicates the maximum number of atxs a block can reference
```

File: miner/block_builder.go [hunk 13]
```diff
@@ -120,6 +126,8 @@ func NewBlockBuilder(minerID types.NodeId, sgn Signer, net p2p.Service, beginRou
 
 }
 
+// Start starts the process of creating a block, it listens for txs and atxs received by gossip, and starts querying
+// blopck oracly chen it should create a block. this function returns an error if Start was already called once
```
Comment (noamnelke) [inline miner/block_builder.go]:
```suggestion
// block oracle when it should create a block. This function returns an error if Start was already called once
```

File: miner/block_builder.go [hunk 14]
```diff
@@ -159,8 +168,8 @@ type meshProvider interface {
 	GetBlock(id types.BlockID) (*types.Block, error)
 }
 
-//used from external API call?
-func (t *BlockBuilder) AddTransaction(tx *types.Transaction) error {
+//used from external API call to dd transcrtion
```
Comment (noamnelke) [inline miner/block_builder.go]:
```suggestion
// used from external API call to add transaction
```

File: miner/tx_pool.go [hunk 15]
```diff
@@ -24,6 +26,7 @@ func NewTxMemPool() *TxMempool {
 	}
 }
 
+// Get returns transaction by provided id id, it returns an error it transaction is not found
```
Comment (noamnelke) [inline miner/tx_pool.go]:
```suggestion
// Get returns transaction by provided id, it returns an error if transaction is not found
```

File: miner/tx_pool.go [hunk 16]
```diff
@@ -34,6 +37,7 @@ func (t *TxMempool) Get(id types.TransactionId) (*types.Transaction, error) {
 	return nil, errors.New("transaction not found in mempool")
 }
 
+// GetTxIdsByAddress returns all transactions from a specific source address
```
Comment (noamnelke) [inline miner/tx_pool.go]:
```suggestion
// GetTxIdsByAddress returns all transactions from/to a specific address
```

File: miner/tx_pool.go [hunk 17]
```diff
@@ -42,6 +46,8 @@ func (t *TxMempool) GetTxIdsByAddress(addr types.Address) []types.TransactionId
 	return ids
 }
 
+// GetTxsForBlock gets a specific number of random txs for a block. this function also receives a state calculation function
```
Comment (noamnelke) [inline miner/tx_pool.go]:
```suggestion
// GetTxsForBlock gets a specific number of random txs for a block. This function also receives a state calculation function
```

File: miner/tx_pool.go [hunk 18]
```diff
@@ -78,6 +84,7 @@ func getRandIdxs(numOfTxs, spaceSize int) map[uint64]struct{} {
 	return idxs
 }
 
+// Put inserts a transaction into the mem pool. it indexes it by source and dest addresses as well
```
Comment (noamnelke) [inline miner/tx_pool.go]:
```suggestion
// Put inserts a transaction into the mem pool. It indexes it by source and dest addresses as well
```

File: state/account_state.go [hunk 19]
```diff
@@ -8,6 +8,7 @@ import (
 	"math/big"
 )
 
+// AccountState is the interface defined to auery a single account state
```
Comment (noamnelke) [inline state/account_state.go]:
```suggestion
// AccountState is the interface defined to query a single account state
```

File: state/account_state.go [hunk 20]
```diff
@@ -18,13 +19,16 @@ type AccountState interface {
 	GetAddress() types.Address
 }
 
+// StateObj is the struct in which account information is stored. it contains account info such as nonce and balane
```
Comment (noamnelke) [inline state/account_state.go]:
```suggestion
// StateObj is the struct in which account information is stored. It contains account info such as nonce and balance
```

File: state/dump.go [hunk 21]
```diff
@@ -24,25 +24,28 @@ import (
 	"github.com/spacemeshos/go-spacemesh/trie"
 )
 
+// DumpAccount is an helper struct that helps dumping account balance and nonce in json form
```
Comment (noamnelke) [inline state/dump.go]:
```suggestion
// DumpAccount is a helper struct that helps dumping account balance and nonce in json form
```

File: state/dump.go [hunk 22]
```diff
@@ -24,25 +24,28 @@ import (
...
 type DumpAccount struct {
 	Balance string `json:"balance"`
 	Nonce   uint64 `json:"nonce"`
 }
 
+// Dump is a sstruct used to dump an entire state root into json form
```
Comment (noamnelke) [inline state/dump.go]:
```suggestion
// Dump is a struct used to dump an entire state root into json form
```

File: state/global_state.go [hunk 23]
```diff
@@ -26,6 +27,8 @@ type GlobalStateDB interface {
 	TrieDB() *trie.Database
 }
 
+// StateDB is the struct that performs logging of all account states. it consists of a state trie that contains all
+// account data in its leaves. it also stores a dirty object list to dump when state is committed into db
```
Comment (noamnelke) [inline state/global_state.go]:
```suggestion
// StateDB is the struct that performs logging of all account states. It consists of a state trie that contains all
// account data in its leaves. It also stores a dirty object list to dump when state is committed into db
```

File: state/processor.go [hunk 24]
```diff
@@ -16,15 +16,19 @@ import (
 	"sync"
 )
 
+// StatePreImages is a struct that contains a root hash and the transactions that are in store of this root hash
 type StatePreImages struct {
 	rootHash  types.Hash32
 	preImages []*types.Transaction
 }
 
+// Projector interface defince the interface for a struct that can project the state aof an account by applying txs from
```
Comment (noamnelke) [inline state/processor.go]:
```suggestion
// Projector interface defines the interface for a struct that can project the state of an account by applying txs from
```

**Review by noamnelke — CHANGES_REQUESTED** [review]
I found a few typos (and one actual mistake), I suggest running the IDE's spell checker on these files.

File: cmd/node/app_test.go [hunk 25]
```diff
@@ -71,7 +71,7 @@ func Test_PoETHarnessSanity(t *testing.T) {
 	require.NotNil(t, h)
 }
 
-func (suite *AppTestSuite) initMultipleInstances(cfg *config.Config, rolacle *eligibility.FixedRolacle, rng *amcl.RAND, numOfInstances int, storeFormat string, genesisTime string, poetClient *activation.HTTPPoetClient, fastHare bool, clock TickProvider, network Network) {
+func (suite *AppTestSuite) initMultipleInstances(cfg *config.Config, rolacle *eligibility.FixedRolacle, rng *amcl.RAND, numOfInstances int, storeFormat string, genesisTime string, poetClient *activation.HTTPPoetClient, fastHare bool, clock TickProvider, network network) {
```
Comment (noamnelke) [inline cmd/node/app_test.go:74]: Isn't this causing an override of the type by the variable? Maybe rename the var to `net`?

File: cmd/node/multi_node.go [hunk 26]
```diff
@@ -44,13 +46,15 @@ func NewManualClock(genesisTime time.Time) *ManualClock {
...
-	t.m.Unlock()
+// Unsubscribe removes this channel ch from channels notified on tick
+func (clk *ManualClock) Unsubscribe(ch timesync.LayerTimer) {
+	clk.m.Lock()
+	delete(clk.subs, ch)
+	clk.m.Unlock()
 }
 
-func (t *ManualClock) StartNotifying() {
+// StartNotifying is empty because this clokc is manual
```
Comment (noamnelke) [inline cmd/node/multi_node.go]:
```suggestion
// StartNotifying is empty because this clock is manual
```

File: cmd/node/node.go [hunk 27]
```diff
@@ -203,7 +210,7 @@ func (app *SpacemeshApp) introduction() {
 	log.Info("Welcome to Spacemesh. Spacemesh full node is starting...")
 }
 
-// this is what he wants to execute Initialize app starts
+// Initialize is what he wants to execute Initialize app starts
```
Comment (noamnelke) [inline cmd/node/node.go]: Let's write something meaningful here

**Review by noamnelke — APPROVED** [review]
Looks good in general. There was one meaningless comment that should be fixed. Can be merged afterwards.

**Comment by antonlerner** [comment]
bors merge

**Comment by antonlerner** [comment]
bors merge

**Pull request closed without merging**